src/OrchardCore.Modules/OrchardCore.ContentFields/Drivers/NumericFieldDisplayDriver.cs

```diff
@@ -40,7 +40,15 @@ public override IDisplayResult Edit(NumericField field, BuildFieldEditorContext
             return Initialize<EditNumericFieldViewModel>(GetEditorShapeType(context), model =>
             {
                 var settings = context.PartFieldDefinition.GetSettings<NumericFieldSettings>();
-                model.Value = context.IsNew ? settings.DefaultValue : Convert.ToString(field.Value, CultureInfo.CurrentUICulture);
+
+                if (field.Value.HasValue)
```

Comment:
why would C#? default value have priory over settings Default?

Reply:
As you have seen I was working on a way to import content items from other sources like Excel file. at that time, the value is set prior the driver is called for rendering. The content item isn't yet created but will have some default values.